### PR TITLE
add tentative support for both long & short option examples

### PR DESCRIPTION
**Pull request by blueglyph**
This PR isn't a new page or a page update, but a proposal to change their content.

We suggest to include, when relevant, both long and short format options as described in the modified `style-guide.md`.

So instead of this:

```
- Synchronize and update all packages:

`sudo pacman --sync --refresh --sysupgrade`
```

We would have this:

```
- Synchronize and update all packages:

`sudo pacman --sync --refresh --sysupgrade`
`sudo pacman -Syu`
```

and modified clients would allow the users to select which format they prefer to see (or just see both).

- rules are specified in `contributing-guides/style-guide.md`
- a new client parameter to let the user choose between long, short or both formats is specified in `CLIENT-SPECIFICATION.md`
- the `pages/linux/pacman.md` page has been modified to illustrate the change
- no conflict were identified in the current pages: there is no successive lines containing code example
- one client, tealdeer, has been modified accordingly: https://github.com/blueglyph/tealdeer/releases/tag/CODE-FILTER-RC2

See also discussion in https://github.com/tldr-pages/tldr/issues/8046

### CODE REVIEW

**Comment by blueglyph** [comment]
Nice to see the bot has spotted the double lines. Now I see where the test scripts are, including tldr-lint. What I can do is remove the offending `pacman.md` file and provide it separately.

**Comment by blueglyph** [comment]
Better. It still shows I haven't signed the CLA, though I did. I suppose it takes some time to synchronize. Otherwise let me know if I did something wrong there.

**Comment by EmilySeville7cfg** [comment]
@blueglyph, to make it work, revert changes made to `pacman` (because now our linter will not allow you to merge such changes). The linter should be modified at the first place to make CI checks pass. :)

**Comment by blueglyph** [comment]
@EmilySeville7cfg I've done it but I have made a mistake and reverted everything (not used to the UI I used). I've made a new commit to fix it, but it doesn't seem to be seen here.

How should I update the pull request with my last commit?

File: contributing-guides/style-guide.md
Comment (EmilySeville7cfg) [inline contributing-guides/style-guide.md:118]:
```suggestion
- Both **GNU-style long** (like `--help`) and **short** (like `-h`) option formats must be shown in example distributed to two lines, each starting and ending with a backtick.
- The first example's line must contain **long** cross-platform compatible options (intended to work the same across multiple platforms).
- The second one is for short options.
- If some **long** option has no short counterpart or vice versa then just existing option must be present in both example's lines.
- If both example's lines are equal just one must be leaved.
```

Optional fix: broken nesting: nesting can be reduced.

**Review by EmilySeville7cfg — APPROVED** [review]
No mandatory fixes found, approved.
Thanks for a contribution! 🚀

**Review by sbrl — CHANGES_REQUESTED** [review]
I think this is overcomplicating things. tldr-pages strength is in it's simplicity: minimal parsing of pages is required. The `--option-format`/`-O` seems overcomplicated and unnecessary.

Displaying both long and short option versions of examples also seems unnecessary. In tldr-pages the general guidance is that you should use whatever seems most appropriate at the time. For example, on the page `ls` or `tar` I would be inclined to use short options in examples, whereas in the page for a command such as `lvresize` (part of LVM), I would use long options since it's a command that isn't likely to be used as often.

**Comment by blueglyph** [comment]
> Displaying both long and short option versions of examples also seems unnecessary. In tldr-pages the general guidance is that you should use whatever seems most appropriate at the time. For example, on the page `ls` or `tar` I would be inclined to use short options in examples, whereas in the page for a command such as `lvresize` (part of LVM), I would use long options since it's a command that isn't likely to be used as often.

That discussion already took place actually (see link in the first post), and the general conclusion was in favour of short options or at least both. That's probably why pacman was changed to short. So it's necessary to some users at least.

Most people use the short options, and if you're looking at discussions for ex. about pacman, you'll see that it's mostly always short options. Besides, since they're less easy to remember, that's where the tool is really useful. But the choice between both can't hurt either.

The option `--option-format` could be removed if it's offending. It's just there to make it easier to switch from one tool to another.

**Comment by EmilySeville7cfg** [comment]
>  In tldr-pages the general guidance is that you should use whatever seems most appropriate at the time. 

There is formal [rule](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax) to decide when to use each options style. :) To make a long story short: long options are used just when they are portable. No exceptions are done.

**Comment by EmilySeville7cfg** [comment]
@blueglyph, I am leaving this project to create a better project-replacement for TlDr. I invite you here and we can build smth better together much faster.

**Comment by blueglyph** [comment]
The status is
- **1 change requested**, though I see no clear request from @sbrl, just the reopening of a discussion (I suggest to continue any discussion in the original PR, if that's really necessary). Or could you clarify the change you'd like to see?
- **license/cla** pending, though I have signed the CLA (and did again, just to make sure), so I don't see how to fix that

**Comment by blueglyph** [comment]
I'm still waiting for an answer from @sbrl (or his retracting if there are no change request) and for another review.

**Comment by EmilySeville7cfg** [comment]
@blueglyph, can u update official clients too to support this syntax proposal? I guess it's the other reason why @sbrl declined this changed: as just one client (and not official) supports such syntax.

**Comment by blueglyph** [comment]
@EmilySeville7cfg I have updated one client, and if you remember our discussion on the link I gave above, it was working well. I even made a binary release for Windows and Linux. I'm not sure what you mean by "official", but this one certainly counts as such, since it's a very good implementation - and in Rust, which makes it fast, safe and small.

Unfortunately I cannot possibly do a pull request until the specs are changed first. This has to be the first step before any real "official" implementation. And since it has been a while, I'm not even sure it's possible without a complicated merge, so the sooner this is resolved, the better. 🙂

sbrl didn't mention any of this so I don't think that's the reason. It's a change request without specifying what should be changed - seeing his/her previous comment and the obvious lack of reply, I suspect it's simply a way to block the proposition. It's unfortunate since it has been discussed elsewhere and people seemed favourable.

So to be clear: I will _not_ modify another client, nor do anything else until this bit here is clarified (which seems unlikely).

**Comment by EmilySeville7cfg** [comment]
@blueglyph, by official clients I mean Python and NodeJS ones which are recommended in the main README.

Yep, I agree, that changes were requested maybe not to just rethink some syntax, but maybe to decline this PR completely. Some of my PRs were blocked by her with suggestions to avoid some new style guide rules (which have been removed then, like ranges 1..10). So it's expectable. 

@sbrl definitely has another opinion on what TlDr should be. She prefers simplicity over some rich syntax. It makes sence, but on the same time it limits TlDr. My opinion is not to limit syntax to simple constructs and delegate more tasks for page renders. But... I don't think this project will change the current situation about this soon. So... here's a question: is there any reason for TlDr to exist if it doesn't want to evolve (you know what I mean)? I wanna maintainers to think about this question.

**Comment by blueglyph** [comment]
@EmilySeville7cfg Thanks! I see the clients now, and they're under the same GitHub owner, which makes sense.

To each their own opinion, what's important is to respect others' and to take into account what most users want. I was trying to help in that direction, especially since I saw that now some of the pages had the short format only, which is inconsistent. Besides, the proposition included an option to choose, and it would likely have been a few pages only.

So if that's really the motive behind it, I can only pity that person for such pettiness. All it will achieve is put more users off and scare potential contributors away.

BTW, I see there are many pending pull requests here, a good share of those apparently just waiting for a review for a long time. There would probably be even more if the bot wasn't closing them, disregarding whatever work has been done. So I'll just let this die peacefully, I gave it a chance but I hate wasting time. 😅

**Comment by blueglyph** [comment]
I deleted the fork to make some free space, since it was hopeless.

**Pull request closed without merging**
